firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Driver/I2C/SFM3019/Device.txt

```diff
@@ -0,0 +1,253 @@
+
+Scenario: SFM3019 Device:: SFM3019 device behaves correctly when request_conversion_factors function is called 
+  GIVEN('A SFM3019 device constructed with a mock I2C device , global I2C device has write buffer '0x01' and gas parameter') 
+    WHEN('The request_conversion_factors function is called with gas type O2 as input parameter') 
+      THEN('The request_conversion_factors function returns ok status') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor O2 command bytes '0x3661 followed by '0x36033A'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged ') 
+
+    WHEN('The request_conversion_factors function is called with  gas type air as input parameter') 
+      THEN('The request_conversion_factors function returns ok status') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor air command bytes '0x3661 followed by '0x3608D0'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN(
+        'The request_conversion_factors function is called with gas type mixture as input parameter') 
+      THEN('The request_conversion_factors returns ok') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor mixture command byte '0x3661 followed by '0x3632CE'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+Scenario: SFM3019 Device:: SFM3019 device behaves correctly when read_conversion_factors function is called 
+  GIVEN('A SFM3019 device constructed with a mock I2C device having read buffer 0x[ae dc 6a 00 00 81 00 00 81], global I2C device has write buffer '0x01' and air as gas parameter') 
+    WHEN(The read_conversion_factors function is called with a default-initialized ConversionFactors output parameter and with the read buffer of mock I2C device consisting 0x[01 02], corresponding to scale factor 10 and offset of -5') 
+      THEN('The read_conversion_factors return ok status') 
+      THEN('The output parameter scale factor is 10')
+      THEN('The output parameter offset is -5')
+      THEN('The mock_device I2C's write buffer remains unchanged') 
+      THEN('The mock I2C's read buffer changes to initially pushed 0x[01 02] bytes') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN(The read_conversion_factors function is called with a default-initialized ConversionFactors output parameter and with the read buffer of mock I2C device consisting 0x[01 02], corresponding to scale factor 170 and offset of -24576') 
```

Comment:
It would be helpful to also have another version of this WHEN where the ConversionFactors output parameter has random numbers inside it; the THENs should all be the same as the THENs in this WHEN.